### DISCUSSION TITLE
Txt to html

### DISCUSSION THREAD

**Discussion by MCSUU** [Tyrrrz/DiscordChatExporter#1047] [Q&A]
Can i change my txt chat save to html chat save? (i cant do another save bcs account is disabled)

**Comment by Yudi** [comment]
It isn't possible to convert already exported files to another format. :-(
You'd have to export the chat again.

**Reply by Rajdave69** [reply]
Someone could probably write a conversion script, pretty sure each format has all the required info to convert to another format (except profile pictures perhaps)

**Reply by 96-LB** [reply]
TXT format is lacking in most data. HTML and JSON have the most information but the other methods are deprecated.

**Reply by Stoxis** [reply]
I created a script that converts the HTML channel exports into the JSON format https://gist.github.com/Stoxis/a3f5ce5b36732048749dbab8f90ac546

The way I set up the category and message IDs is wrong but works perfectly fine, if anybody knows how to re-generate the correct IDs with the limited information provided in the HTML file format please do tell me.

**Reply by The-TallGuy** [reply]
what are the bs4, tqdm and requests packages?

**Reply by Yudi** [reply]
[beautifulsoup4](https://pypi.org/project/beautifulsoup4/), [tqdm](https://pypi.org/project/tqdm/), and [requests](https://pypi.org/project/requests/) respectively

`pip install beautifulsoup4 tqdm requests`

**Reply by The-TallGuy** [reply]
Thank you, the script runs now.
There appears to be another issue though. Apparently there is a div with the class info__guild-name that has no text, but the script expects it to have some. Is there a way I could fix that?

**Reply by The-TallGuy** [reply]
Anyone? Please?

**Reply by 96-LB** [reply]
@The-TallGuy You'll have to edit the script yourself. You could perhaps do something like:
```py
# replace:
guild_name = soup.find('div', class_='info__guild-name').text

# with:
try:
    guild_name = soup.find('div', class_='info__guild_name').text
except AttributeError:
    guild_name = None
```
But it looks like certain parts of the script expect there to be a real guild name, so you might have to make other modifications. You could also manually edit your HTML file and put text of your choice in the info__guild-name div.

Disclaimer: I have not tried the script.

**Reply by The-TallGuy** [reply]
Thank you. Will try. Didn't think about just trying to use an exception for some reason.
